### ISSUE TITLE
Template Support

### ISSUE THREAD

**Issue by JiffyRob** [bitcraft/pytmx#171]
It looks like pytmx can't load [templates](https://doc.mapeditor.org/en/stable/manual/using-templates/#).  

When an object with a template is loaded, I see a `TiledObject.template` attribute which corresponds to the path of the template's `.tx` file (relative to the path of the map), but there does not seem to be a way to get any meaningful data out of it without parsing the file separately, which would more or less ruin the point of this library's existence.  

The desired behavior would either be replacing the `TiledObject.template` attribute with a `TiledTemplate` object or similar, or straight up overriding the object's attributes (most intuitive, in my opinion).

Theoretically, adding support would not be too difficult.  I'd be willing to try an implement it myself if necessary, but this seems to be a pretty simple feature to not have around.  Am I missing something?

Here's [my current tiled project](https://github.com/bitcraft/pytmx/files/10473538/tiled-project.zip) as a test case.  The assets are from [here](https://pixel-boy.itch.io/ninja-adventure-asset-pack) (CC0).

**Comment by JiffyRob** [comment]
I found a workaround - after you create the object and set your attributes, right click it and hit 'detach'.  This will decouple it from the template and load all the inherited attributes directly.

I still think native support would be a good idea, though.  Or at the very least a note should be put in the docs.

**Comment by drvolk** [comment]
I see it also like this, pytmx should provide a way to load the data from "TiledObject.template".
I currently have the problem that I have used such templates for the definition of TileCollider shapes. 
Unfortunately pytmx does not take over the values for height and width from the template definition when initializing the TileCollider objects, so that these values are 0 and e.g. a correspondingly created Collider Rect() has no effect.
Instead I have to load the ".tx" file from the "template" property itself and parse it to get the correct height and width of the Collider object for creating a corresponding Rect().

**Comment by Charles-ai-hue** [comment]
In Tiled, if an object's property is **unmodified** (greyed out), Tiled only saves the `template` reference in the object tag. 
To bypass this, **temporarily modify every greyed-out property** , save, then revert the changes and save again.   
After this, Tiled will embed **all properties directly in the object tag**, even if they match the template. Now `pytmx` can read them without needing the `.tx` file.

**Comment by ghost** [comment]
Yes this bit me too recently. Spent too many hours debugging my code only to realise it's a pytmx issue.
 I wrote a method.

Called by:

```
if obj.template != None:
                self.load_template(obj)
```

```
def load_template(self, obj):
        '''
        Load a template file and pass it as an object back
        @param template filename
        @return an object
        '''
        try:
            tree = ET.parse(os.path.join("src","map",obj.template))
            root = tree.getroot()
            DEBUG(root[0].attrib)
            obj.name = root[0].attrib['name']
            obj.type = root[0].attrib['type']
            for prop in root.iter('property'):    
                # Ony set the template value if it's not set in the instance. Do not overwrite.
                # e.g. <property name="price" type="int" value="20"/>
                # Check if a property value in not already set
                if obj.properties.get(prop.attrib.get('name')) == None: # So  we can set it here
                    if prop.attrib.get('type') == 'int': # Need to convert to int
                        # {'name': 'price', 'type': 'int', 'value': '4'}
                        obj.properties[prop.attrib.get('name')] = int(prop.attrib.get('value'))
                    else:
                        obj.properties[prop.attrib['name']] = prop.attrib.get('value')
                    # Each prop.attrib is a dictionary
                else:
                    DEBUG(f"Not setting {prop.attrib.get('name')}")
```
Clearly the comments and DEBUG's  may not be relavent but may supply an idea to deal with templates.

Obvioulsy a generic version needs to be incorporated into pytmx.
Hope this helps someone.
